### PR TITLE
add copy overload to IO to copy at most *limit* bytes

### PR DESCRIPTION
**Pull request by jreinert**

### CODE REVIEW

**Comment by asterite** [comment]
This is useful, and Ruby also [has this](http://ruby-doc.org/core-2.2.3/IO.html#method-c-copy_stream). Give us some time to review it with @waj, and also check #1641, and we'll probably merge it.

**Comment by asterite** [comment]
@jreinert Do you want to rebase this? Then we'll merge it.

**Comment by jreinert** [comment]
@asterite done!

File: src/io.cr [hunk 1]
```diff
@@ -896,6 +896,26 @@ module IO
...
+  # ```
+  # io = StringIO.new "hello"
+  # io2 = StringIO.new
+  #
+  # IO.copy io, io2, 3
+  #
+  # io2.to_s #=> "hel"
+  # ```
+  def self.copy(src, dst, limit)
+    buffer :: UInt8[1024]
```
Comment (asterite) [inline src/io.cr]: @jreinert This should be `buffer = uninitialized UInt8[1024]`

**Comment by miketheman** [comment]
#triage 
@jreinert It seems like there's a need for a syntax update, as pointed out by @asterite .
A rebase against master is always a good idea, as this will trigger the most current CI build/test suite.

**Comment by jreinert** [comment]
all done

**Comment by miketheman** [comment]
@jreinert Thanks for the rebase. As you can see, the build fails on `undefined constant StringIO`.

In crystal 0.9.0, [StringIO was renamed to MemoryIO](https://github.com/crystal-lang/crystal/commit/9b8e6c7e5f35b62503cd1507b1097d6c20c398dd) - could you please rename the specs and docs?

**Comment by asterite** [comment]
Merged in https://github.com/crystal-lang/crystal/commit/5fa22e6a1ddcf0f9e816aea00c6bdb5efb96921d . Thank you!

**Pull request closed without merging**